src/Tools/dotnet-monitor/Egress/S3/MultiPartUploadStream.cs

```diff
@@ -75,6 +117,11 @@ public override void SetLength(long value)
     }
 
     public override async ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken)
+    {
+        await WriteAsync(buffer, false, cancellationToken);
+    }
+
+    public async ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, bool allowPartialWrite, CancellationToken cancellationToken)
```

Comment:
This `allowPartialWrite` parameter seems unnecessary; that's the purpose of the `FinalizeAsync` method.